lib/active_fedora/solr_indexing_strategy.rb

```diff
@@ -0,0 +1,26 @@
+module ActiveFedora
+  ##
+  # An application strategy to also apply solr indexes for each property.
+  # @note If how a field is indexed changes based on property, this would be a
+  #   good place to define that.
+  class SolrIndexingStrategy
+    # @param [Array<Symbol>] index_types Ways to index all properties.
+    def initialize(index_types=[])
+      @index_types = Array(index_types)
+    end
+
+    # @param [ActiveFedora::Base] object The object to apply the property to.
+    # @param [ActiveTriples::Property, #name, #to_h] property The property to define.
+    def apply(object, property)
+      object.property property.name, property.to_h do |index|
+        unless index_types.empty?
```

Comment:
If you encapsulate this inner block in a method, it would be easier to switch conditionally (based on property.name).